controllers/strong_consistency.go

```diff
@@ -0,0 +1,347 @@
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *SingleClusterReconciler) runRecluster(hostConns []*deployment.HostConn, policy *as.ClientPolicy) error {
+	r.Log.Info("Run recluster")
+
+	for _, hostConn := range hostConns {
+		if err := r.recluster(hostConn, policy); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *SingleClusterReconciler) validateClusterState(policy *as.ClientPolicy) error {
+
+	hostConns, err := r.newAllHostConn()
+	if err != nil {
+		return err
+	}
+
+	scNsList, err := r.getSCNamespaces(hostConns, policy)
+	if err != nil {
+		return err
+	}
+
+	for _, ns := range scNsList {
+		if err := r.validateClusterNsState(hostConns, policy, ns); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *SingleClusterReconciler) validateClusterNsState(hostConns []*deployment.HostConn, policy *as.ClientPolicy, ns string) error {
+	r.Log.Info("Validate Cluster namespace State. Looking for unavailable or dead partitions", "namespaces", ns)
+
+	for _, hostConn := range hostConns {
+
+		kvMap, err := r.getNamespaceStats(hostConn, policy, ns)
+		if err != nil {
+			return err
+		}
+
+		// https://docs.aerospike.com/reference/metrics#unavailable_partitions
+		// This is the number of partitions that are unavailable when roster nodes are missing.
+		// Will turn into dead_partitions if still unavailable when all roster nodes are present.
+		// Some partitions would typically be unavailable under some cluster split situations or
+		// when removing more than replication-factor number of nodes from a strong-consistency enabled namespace
+		if kvMap[nsKeyUnavailablePartitions] != "0" {
+			return fmt.Errorf("cluster namespace %s has non-zero unavailable_partitions %v", ns, kvMap[nsKeyUnavailablePartitions])
+		}
+
+		// https://docs.aerospike.com/reference/metrics#dead_partitions
+		if kvMap[nsKeyDeadPartitions] != "0" {
+			return fmt.Errorf("cluster namespace %s has non-zero dead_partitions %v", ns, kvMap[nsKeyDeadPartitions])
+		}
+	}
+	return nil
+}
+
+func (r *SingleClusterReconciler) isNamespaceSCEnabled(policy *as.ClientPolicy, hostConn *deployment.HostConn, ns string) (bool, error) {
+
+	cmd := fmt.Sprintf("get-config:context=namespace;id=%s", ns)
+
+	res, err := hostConn.ASConn.RunInfo(policy, cmd)
+	if err != nil {
+		return false, err
+	}
+
+	configs, err := ParseInfoIntoMap(res[cmd], ";", "=")
+	if err != nil {
+		return false, err
+	}
+	scStr, ok := configs[nsKeyStrongConsistency]
+	if !ok {
+		return false, fmt.Errorf("strong-consistency config not found, config %v", res)
+	}
+	scBool, err := strconv.ParseBool(scStr)
+	if err != nil {
+		return false, err
+	}
+	r.Log.Info("Check if namespace is SC namespace", "ns", ns, nsKeyStrongConsistency, scBool)
+
+	return scBool, nil
+}
+
+// Info calls
+
+func (r *SingleClusterReconciler) recluster(hostConn *deployment.HostConn, policy *as.ClientPolicy) error {
+	cmd := "recluster:"
+	res, err := hostConn.ASConn.RunInfo(policy, cmd)
```

Comment:
lets use recluster from management lib. Should we move all functions calling host info commands to management lib.